libraries/AP_HAL_ChibiOS/CANFDIface.cpp

```diff
@@ -996,13 +992,10 @@ uint32_t CANIface::getErrorCount() const
 }
 
 #if CH_CFG_USE_EVENTS == TRUE
-ChibiOS::EventSource CANIface::evt_src_;
-bool CANIface::set_event_handle(AP_HAL::EventHandle* handle)
+bool CANIface::set_event_handle(AP_HAL::CountingSemaphore *handle)
 {
-    CriticalSectionLocker lock;
```

Comment:
check if handle is null, maybe assert and return false as well.

Reply:
why is it invalid to set a null handle?

Reply:
Yeah, we already check before using everywhere so doesn't matter.